### PR TITLE
Add udpreceiver

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:** 

Add [udpreceiver](https://github.com/open-telemetry/opentelemetry-log-collection/blob/main/docs/operators/udp_input.md) from log-collection repository

**Link to tracking Issue:**
N/A

**Testing:**
Unit tests

**Documentation:**
Readme

### CODE REVIEW

**Comment by djaglowski** [comment]
This receiver hasn't been formally proposed as far as I know, so I think we need to validate some basic assumptions:
- The use case is to receive _non-syslogs_ over udp, correct?
- Assuming yes, then let's validate with maintainers that this is ok to have in the contrib repo. //cc @tigrannajaryan 
- Assuming yes, then are we ok with the name, or should this perhaps be `udplogreceiver`?

**Comment by tigrannajaryan** [comment]
Is this a widely known log protocol? I can't tell from https://github.com/open-telemetry/opentelemetry-log-collection/blob/main/docs/operators/udp_input.md

File: receiver/udplogreceiver/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# UDP Receiver
+
+Receives logs from udp using
+the [opentelemetry-log-collection](https://github.com/open-telemetry/opentelemetry-log-collection) library.
```
Comment (bogdandrutu) [inline receiver/udplogreceiver/README.md:4]: Please explain the protocol/usages since probably that udp library part will be replaced by this.

**Review by bogdandrutu — COMMENTED** [review]
/cc @djaglowski

**Comment by sumo-drosiek** [comment]
@bogdandrutu  sorry for mentioning it in this thread. Discussion about analogical receiver for tcp is under #3190

**Comment by tigrannajaryan** [comment]
@sumo-drosiek I assume this needs updates similar to tcpreceiver, right?

**Comment by sumo-drosiek** [comment]
@tigrannajaryan yes, but it's going to be more complicated (especially multiline detection) due to being non-connection oriented

File: receiver/udplogreceiver/README.md (hunk 1 above)
Comment (sumo-drosiek) [reply receiver/udplogreceiver/README.md:4]: Could you give some example please?
I'm not exactly sure what are you asking for

**Comment by djaglowski** [comment]
[opentelemetry-log-collection #127](https://github.com/open-telemetry/opentelemetry-log-collection/pull/127) was recently merged and should support this PR.

**Comment by sumo-drosiek** [comment]
@djaglowski I will update this altogether with hostname resolution support

**Comment by sumo-drosiek** [comment]
- I bumped opentelemetry-log-collection to latest main
 - updated tcplog readme
 - updated udplog readme

**Comment by sumo-drosiek** [comment]
I got
```
windows-test - Unauthorized
loadtest - Unauthorized
```
:scream:

**Comment by jpkrohling** [comment]
I restart the CI builds, just in case...

**Comment by tigrannajaryan** [comment]
What's `unauthorized`? Did we change anything in the build setup recently?

**Comment by jpkrohling** [comment]
Does this ring a bell? From https://circleci.com/docs/2.0/contexts/#running-workflows-with-a-restricted-context

>To invoke a job that uses a restricted context, a user must be a member of one of the security groups for the context and must sign up for CircleCI. If the user running the workflow does not have access to the context, the workflow will fail with the Unauthorized status.

**Comment by djaglowski** [comment]
All checks [ran and passed 11 days ago](https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-collector-contrib/13473/workflows/01de23be-ece8-4178-b283-01f415c4cc5e), but `.circleci/config.yml` was [updated 5 days ago](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3381) with changes that appear to be related to @jpkrohling's point.

**Comment by djaglowski** [comment]
@owais, do you mind weighing in on this? 

#3381 makes sense to me in general, except I can't quite figure why `windows-test` and `loadtest` require `context: [github-release-and-issues-api-token]` while the other standard CI checks do not. Is this intentional? If so, is it expected that contributors generally have permissions to this context?

**Comment by tigrannajaryan** [comment]
This https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/3423 should solve it, let's wait for it.

**Comment by tigrannajaryan** [comment]
Please rebase from main.

**Comment by jpkrohling** [comment]
Load test `TestLog10kDPS/syslog-udp` failed with:

```
2021/05/20 18:11:05 CPU consumption is 106.3%, max expected is 80%
    test_case.go:334: CPU consumption is 106.3%, max expected is 80%
```

And `TestLog10kDPS/udp - tests` failed with:

```
2021/05/20 18:12:46 CPU consumption is 92.3%, max expected is 80%
    test_case.go:334: CPU consumption is 92.3%, max expected is 80%
```

I'll trigger the tests again, but does look suspicious.

File: testbed/tests/log_test.go [hunk 2]
```diff
@@ -105,7 +105,7 @@ func TestLog10kDPS(t *testing.T) {
 			sender:   datasenders.NewSyslogWriter("udp", testbed.DefaultHost, testbed.GetAvailablePort(t), 1),
 			receiver: testbed.NewOTLPDataReceiver(testbed.GetAvailablePort(t)),
 			resourceSpec: testbed.ResourceSpec{
-				ExpectedMaxCPU: 80,
+				ExpectedMaxCPU: 200,
```
Comment (djaglowski) [inline testbed/tests/log_test.go:108]: Are we able to identify what part of this change has caused the additional resource utilization?

Comment (sumo-drosiek) [reply testbed/tests/log_test.go:108]: I found this: https://github.com/open-telemetry/opentelemetry-log-collection/pull/161

**Review by djaglowski — APPROVED** [review]

**Pull request merged**